src/main/java/io/jenkins/plugins/view/calendar/service/CronWrapper.java

```diff
@@ -0,0 +1,73 @@
+package io.jenkins.plugins.view.calendar.service;
+
+import hudson.scheduler.CronTab;
+import io.jenkins.plugins.extended_timer_trigger.CronTabWrapper;
+import java.time.ZonedDateTime;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import org.jenkinsci.plugins.parameterizedscheduler.ParameterizedCronTab;
+
+public abstract class CronWrapper<T> {
+    public abstract Calendar ceil(long timeInMillis);
+    public abstract Calendar floor(long timeInMillis);
+    public abstract T getCronTab();
+
+    public static class ClassicCronTab extends CronWrapper<CronTab> {
+        private final CronTab cronTab;
+
+        public ClassicCronTab(CronTab cronTab) {
+            this.cronTab = cronTab;
+        }
+
+        @Override
+        public Calendar ceil(long timeInMillis) {
+            return cronTab.ceil(timeInMillis);
+        }
+
+        @Override
+        public Calendar floor(long timeInMillis) {
+            return cronTab.floor(timeInMillis);
+        }
+
+        @Override
+        public CronTab getCronTab() {
+            return cronTab;
+        }
+    }
+
+    public static class ParameterizedCronWrapper extends ClassicCronTab {
```

Comment:
Currently parameterized-scheduler has no useful java api.
https://github.com/jenkinsci/parameterized-scheduler-plugin/pull/294 adds such an api and that would avoid that calendar-view needs to parse the crontab itself. While it could also directly use the ClassicCronTab for now, I had #108 in mind and then the distinction is anyway needed.